library/Network/IPFS/Pin.hs

```diff
@@ -46,21 +57,49 @@ rm cid = ipfsUnpin cid False >>= \case
         logDebug <| "Pinned CID " <> display cid'
         return <| Right cid'
 
-      _ ->
-        logLeft <| UnexpectedOutput <| UTF8.textShow cids
+      _ -> do
+        formattedErr <- parseUnexpectedOutput <| UTF8.textShow cids
+        return <| Left <| formattedErr
 
   Left _ -> do
     logDebug <| "Cannot unpin CID " <> display cid <> " because it was not pinned"
     return <| Right cid
 
-logLeft ::
+-- | Parse and Log the Servant Client Error returned from the IPFS Daemon
+parseClientError ::
+  ( MonadRIO        cfg m
+  , HasLogFunc      cfg
+  )
+  => ClientError
+  -> m (Error)
+parseClientError err = do
+  logError <| displayShow err
+  let newError = case err of
+        (FailureResponse _ response) -> do
+          response
+          |> responseBody
+          |> decode
+          |> \case
+            Just (IPFSErrorBody {message}) ->
+              IPFSDaemonErr <| UTF8.textShow message
+
+            _ ->
+              UnexpectedOutput <| UTF8.textShow err
+
+        unknownClientError ->
+          UnknownAddErr <| UTF8.textShow unknownClientError
+
+  return newError
+
+-- | Parse and Log unexpected output when attempting to pin
+parseUnexpectedOutput ::
   ( MonadRIO cfg m
   , HasLogFunc cfg
-  , Show a
   )
-  => a
-  -> m (Either IPFS.Add.Error b)
-logLeft errStr = do
-  let err = UnknownAddErr <| UTF8.textShow errStr
-  logError <| display err
-  return <| Left err
+  => Text
+  -> m (IPFS.Add.Error)
+parseUnexpectedOutput errStr = do
+  let baseError = UnexpectedOutput errStr
+  let err = UnknownAddErr <| UTF8.textShow <| baseError
```

Comment:
💅  No need to change, but you don't need the multiple `let`s

```haskell
let baseError = UnexpectedOutput errStr
    err = UnknownAddErr <| UTF8.textShow <| baseError
```

Or, if you like lots of consistency when you lets get bigger:

```haskell
let
  baseError = UnexpectedOutput errStr
  err = UnknownAddErr <| UTF8.textShow <| baseError
```